pilot/pkg/model/push_context.go

```diff
@@ -635,9 +676,96 @@ func (ps *PushContext) InitContext(env *Environment) error {
 	if err = ps.initSidecarScopes(env); err != nil {
 		return err
 	}
+	return err
+}
+
+func (ps *PushContext) updateFromExistingContext(env *Environment, oldPushContext *PushContext, updateReq *UpdateRequest) error {
+
+	var err error
+	servicesChanged := false
+	virtualServicesChanged := false
+	destinationRulesChanged := false
+	authZChanged := false
+	envoyFiltersChanged := false
+	sidecarsChanged := false
```

Comment:
Probably the whole logic could be simplified as - so that we do not have iterate the map and keep track of booleans

```
	if _, ok := ConfigTypesUpdated[ServiceEntry.Type]; ok {
		// Logic for Service changed
	} else {
		// copy all the relevant data structures from old push context
	}
```